File: dataset/output/OPENAI_gpt-4o/Task34_OPENAI_gpt-4o.cpp
#include <iostream>
#include <cmath>
#include <string>

bool isNarcissistic(int number) {
    std::string numStr = std::to_string(number);
    int power = numStr.length();
    int sum = 0;

    for (char c : numStr) {
        int digit = c - '0';
        sum += std::pow(digit, power);
    }

    return sum == number;
}

int main() {
    std::cout << isNarcissistic(153) << std::endl;  // 1 (true)
    std::cout << isNarcissistic(1652) << std::endl;  // 0 (false)
    return 0;
}

File: dataset/output-bk/OPENAI_gpt-4o/Task34_OPENAI_gpt-4o.java
// Java Code
public class Main {
    public static boolean isNarcissistic(int number) {
        int originalNumber = number;
        int sum = 0;
        int numberOfDigits = String.valueOf(number).length();
        while (number > 0) {
            int digit = number % 10;
            sum += Math.pow(digit, numberOfDigits);
            number /= 10;
        }
        return sum == originalNumber;
    }

    public static void main(String[] args) {
        int num = 153; // example number
        System.out.println(isNarcissistic(num));
    }
}